gapis/api/templates/vulkan_gfx_api_extras.tmpl

```diff
@@ -577,6 +577,111 @@ bool Vulkan::replayGetEventStatus(Stack* stack, bool pushReturn) {
     }
 }
 ¶
+bool Vulkan::replayGetSemaphoreCounterValue(Stack* stack, bool pushReturn) {
+   auto expected = stack->pop<uint64_t>();
+   auto pValue = stack->pop<uint64_t*>();
+   auto semaphore = stack->pop<uint64_t>();
+   auto device = static_cast<size_val>(stack->pop<size_val>());
+   if (stack->isValid()) {
+        GAPID_DEBUG("vkGetSemaphoreCountValue(%zu, %" PRIu64 ")", device, semaphore);
+        uint32_t return_value;
+        do {
+          return_value = mVkDeviceFunctionStubs[device].vkGetSemaphoreCounterValue(device, semaphore, pValue);
+        } while (*pValue < expected && (return_value == VK_SUCCESS));
```

Comment:
Why this spin, instead of just using `vkWaitSemaphores`?

Reply:
We could do that here. I wanted to be consistent with our VkGetFenceStatus. (Which spins instead of VkWaitForFences, because of hilarious driver bugs).
But if we think this won't actually break I am happy to switch.